cluster/saltbase/salt/kubelet/default

```diff
@@ -102,5 +102,10 @@
   {% set test_args=pillar['kubelet_test_args'] %}
 {% endif -%}
 
+{% set privileged = pillar.get('allow_privileged', false)  -%}
+{% if pillar.get('e2e_test_environment', '').lower() == 'true' -%}
+  {% set privileged = "True" -%}
```

Comment:
> Can you really not do this without such invasive changes?

No, PersistentVolume, NFS, Gluster, Ceph, and iSCSI tests need `--allow-privileged=True` to work. It seems it's really necessary to have two separate test environments. I can rename `ENABLE_E2E_TEST_ENVIRONMENT` to anything else, still, it will probably result in a new Jenkins environment.